test/e2e/util_test.go

```diff
@@ -688,3 +688,27 @@ func assertDeletedWaitForCleanup(t *testing.T, cl client.Client, thing client.Ob
 		t.Fatalf("Timed out waiting for %s to be cleaned up: %v", thing.GetName(), err)
 	}
 }
+
+// getRouteHost returns the host name of the given route for the named
+// IngressController.  If the IngressController has not added its host name to
+// the route's status, this function causes a fatal error.  For simplicity, this
+// function does not check the "Admitted" status condition, so it will return
+// the host name if it is set even if the IngressController has rejected the
+// route.
+func getRouteHost(t *testing.T, route *routev1.Route, router string) string {
```

Comment:
Would switching [canary.getRouteHost ](https://github.com/openshift/cluster-ingress-operator/pull/965/files#diff-aabcb3468eb7e3e07cce919ec78adc2f6012ed230991bc7f40f62868ada392d7R186) to public be a better option than duplicating?

Or is this intentional because it's just E2E test code?

Reply:
The function is simple and yet specialized enough that it seemed simpler to me to duplicate a few lines rather than convert the other `getRouteHost` into a public interface.